File: java-service/src/main/java/com/claud/analyticore/domain/Job.java
package com.claud.analyticore.domain;

import java.time.Instant;
import java.util.List;

// kguanoluisa, Entidad de dominio Job con resultado de análisis, sin nuevas variables, 2026-07-17
public class Job {

    private String id;
    private String texto;
    private EstadoJob estado;
    private String resultadoSentimiento;
    private List<String> palabrasClave;
    private Instant fechaCreacion;
    private Instant fechaActualizacion;

    public Job() {
    }

    public Job(String id, String texto, EstadoJob estado) {
        this.id = id;
        this.texto = texto;
        this.estado = estado;
    }

    public void marcarCompletado(String sentimiento, List<String> palabrasClave) {
        this.estado = EstadoJob.COMPLETADO;
        this.resultadoSentimiento = sentimiento;
        this.palabrasClave = palabrasClave;
        this.fechaActualizacion = Instant.now();
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getTexto() {
        return texto;
    }

    public void setTexto(String texto) {
        this.texto = texto;
    }

    public EstadoJob getEstado() {
        return estado;
    }

    public void setEstado(EstadoJob estado) {
        this.estado = estado;
    }

    public String getResultadoSentimiento() {
        return resultadoSentimiento;
    }

    public void setResultadoSentimiento(String resultadoSentimiento) {
        this.resultadoSentimiento = resultadoSentimiento;
    }

    public List<String> getPalabrasClave() {
        return palabrasClave;
    }

    public void setPalabrasClave(List<String> palabrasClave) {
        this.palabrasClave = palabrasClave;
    }

    public Instant getFechaCreacion() {
        return fechaCreacion;
    }

    public void setFechaCreacion(Instant fechaCreacion) {
        this.fechaCreacion = fechaCreacion;
    }

    public Instant getFechaActualizacion() {
        return fechaActualizacion;
    }

    public void setFechaActualizacion(Instant fechaActualizacion) {
        this.fechaActualizacion = fechaActualizacion;
    }
}

File: python-service/domain/job.py
# kguanoluisa, Entidad Job y enum EstadoJob para el servicio de submisión, variables EstadoJob, 2026-07-17
from dataclasses import dataclass, field
from datetime import datetime
from enum import Enum
from uuid import uuid4


class EstadoJob(str, Enum):
    PENDIENTE = "PENDIENTE"
    PROCESANDO = "PROCESANDO"
    COMPLETADO = "COMPLETADO"


@dataclass
class Job:
    texto: str
    id: str = field(default_factory=lambda: str(uuid4()))
    estado: EstadoJob = EstadoJob.PENDIENTE
    resultado_sentimiento: str | None = None
    palabras_clave: list[str] | None = None
    fecha_creacion: datetime = field(default_factory=datetime.utcnow)
    fecha_actualizacion: datetime = field(default_factory=datetime.utcnow)

    def marcar_procesando(self) -> None:
        self.estado = EstadoJob.PROCESANDO
        self.fecha_actualizacion = datetime.utcnow()

    def marcar_completado(self, sentimiento: str, palabras_clave: list[str]) -> None:
        self.estado = EstadoJob.COMPLETADO
        self.resultado_sentimiento = sentimiento
        self.palabras_clave = palabras_clave
        self.fecha_actualizacion = datetime.utcnow()
